shed/event_streams.py

```diff
@@ -1267,13 +1349,64 @@ def update(self, x, who=None):
 
 
 class QueryUnpacker(EventStream):
-    def __init__(self, db, child, fill=True):
+    """Unpack Queries from the Query node
+
+    This unpacks the start uids passed down from the Query node and creates
+    a restream of the data for each header
+
+    Parameters
+    -----------
+    db: databroker.Broker instance
+        The databroker to be queried
+    child: EventStream instance
+        The stream to be subscribed to
+    fill: bool, optional
+        Whether or not to fill the documents, defaults to True
+
+    """
+    def __init__(self, db, child, fill=True, **kwargs):
         self.db = db
-        EventStream.__init__(self, child)
+        EventStream.__init__(self, child, **kwargs)
         self.fill = fill
 
     def update(self, x, who=None):
-        name, doc = x
+        name, docs = self.curate_streams(x, False)
+        doc = docs[0]
         if name == 'event':
             return [self.emit(nd) for nd in
                     self.db[doc['data']['hdr_uid']].documents(fill=self.fill)]
+
+
+class split(EventStream):
+    def __init__(self, child, n_streams):
+        self.split_streams = [EventStream(
+            stream_name='Split output-{}'.format(i)) for i in range(n_streams)]
+        EventStream.__init__(self, child=child)
+
+    def update(self, x, who=None):
+        name, docs = self.curate_streams(x, False)
+        return [s.emit((name, doc)) for s, doc in zzip(self.split_streams,
+                                                       docs)]
+
+
+class fill_events(EventStream):
```

Comment:
Not happy about this but it needs to be there. The problem is that we could use `map` to apply `db.fill_event` to each event but then we would loose the start metadata as we would override it with data about the `map`. Conversely, it is kinda ok that we don't have provenance for this since it is a purely database driven consideration and deals with the internals of the `RunEngine` giving us tokens and not data.